### PR TITLE
feat(ci): Initial CI/CD pipeline

### LINKED ISSUE #31: automated CI builds
We should add CI to automatically build Linux and MacOS binaries for Git tags.

### PR DESCRIPTION
**Pull request by anurag-rajawat**
Fixes #31 

To get a preview [click here](https://github.com/anurag-rajawat/mnemonikey/releases)

### CODE REVIEW

**Comment by kklash** [comment]
Thanks very much for this! Got some things I need to do IRL but i'll take a look at this by the end of the week

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        with:
+          go-version: 1.19
+
+      - name: Build
+        run: |
+          cd ./cmd/mnemonikey
+          go build -v ./...
+
+      - name: Test
+        run: go test -v ./...
```
Comment (kklash) [inline .github/workflows/build.yml:29]: Could you please leave the existing `test.yml` action as it was, and remove this `build.yml` action? I think i'd like to keep the testing pipeline we already had. 

Although i do like your change here:

```yml
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
```

I agree that is probably the right approach. Could you please apply that to `test.yml` as a drive-by? :smile:

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.19
+
+      - name: Build
+        run: |
+          cd ./cmd/mnemonikey
+          go build -v ./...
```
Comment (kklash) [inline .github/workflows/build.yml:26]: There's no sense running `go build` unless we cache or upload the resulting binary somewhere. Probably it is better to just run `go vet` and `go test` (or `go test -vet=all` to do both at once)

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+permissions:
+  contents: write
+
+jobs:
+  create-release:
+    name: Release Go Binary
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        goos: [ linux, darwin ]
```
Comment (kklash) [inline .github/workflows/release.yml:15]: Seems like `wanyoucao577/go-release-action` supports windows builds. Can you maybe try adding 'windows' to the build matrix? 

```suggestion
        goos: [ linux, darwin, windows ]
```

(You might also need to exclude windows+arm64 from the matrix)

Even if the windows binaries don't work out-of-the box for everyone, it would probably be better than nothing. If someone has issues running the binary they can open an issue and we can investigate further.

**Review by kklash — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        goarch: [ "386", amd64, arm64 ]
+        exclude:
+          - goarch: "386"
+            goos: darwin
+    steps:
+      - uses: actions/checkout@v3
+      - uses: wangyoucao577/go-release-action@v1
```
Comment (kklash) [inline .github/workflows/release.yml:22]: Oh and could you please use the commit hash of this action instead of the major tag version? This [prevents the maintainer of that action from changing the code in their action to something malicious.](https://julienrenaux.fr/2019/12/20/github-actions-security-risk/). It doesn't guarantee complete safety because downstream sub-dependency actions could still change, but it's better than just a version tag.

```suggestion
      - uses: wangyoucao577/go-release-action@b989099
```

In the future i'd like to make sure all actions are either stored in this repo, or are officially verified github actions. Running 3rd party actions to build security-sensitive software is a risky proposition. This is good for the meantime since Mnemonikey is still in the early testing/beta stages.

File: .github/workflows/build.yml (hunk 1 above)
Comment (anurag-rajawat) [reply .github/workflows/build.yml]: Sure, I'll remove the `build` action and apply the above changes in `test` action.

File: .github/workflows/build.yml (hunk 2 above)
Comment (anurag-rajawat) [reply .github/workflows/build.yml]: I removed the `build` action.

**Review by kklash — APPROVED** [review]
Awesome. Thank you!
